homeassistant/components/xiaomi_aqara.py

```diff
@@ -216,13 +232,37 @@ def device_state_attributes(self):
         """Return the state attributes."""
         return self._device_state_attributes
 
-    def push_data(self, data):
+    @callback
+    def _set_unavailable(self, now):
+        """Set state to UNAVAILABLE."""
+        self._remove_unavailability_tracker = None
+        self._is_available = False
+        self.schedule_update_ha_state()
```

Comment:
This is a sync method called from an async context (the `@callback` annotation). That's not allowed.

Reply:
Replaced with async version.